src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5548,12 +5603,24 @@ private SyntaxToken ParseIdentifierToken(ErrorCode code = ErrorCode.ERR_Identifi
                     return result;
                 }
 
+                bool isFieldOrValueInKeywordContext = IsCurrentTokenFieldOrValueInKeywordContext();
                 SyntaxToken identifierToken = this.EatToken();
 
                 if (this.IsInAsync && identifierToken.ContextualKind == SyntaxKind.AwaitKeyword)
                 {
                     identifierToken = this.AddError(identifierToken, ErrorCode.ERR_BadAwaitAsIdentifier);
                 }
+                else if (isFieldOrValueInKeywordContext) // PROTOTYPE: Are we testing this with "value"?
+                {
+                    if (reportErrorRatherThanConvertingFieldOrValue.GetValueOrDefault())
```

Comment:
not gettnig why we need this `bool?`.  seems out of sorts with how we genrally do contexts and parsing of identifiers/contextual-keywords.